cmd/agent/app/builder.go

```diff
@@ -99,63 +97,36 @@ type HTTPServerConfiguration struct {
 	HostPort string `yaml:"hostPort" validate:"nonzero"`
 }
 
-// WithReporter adds auxiliary reporters.
-func (b *Builder) WithReporter(r reporter.Reporter) *Builder {
```

Comment:
Ran in to an issue here. This needs to stay. We internally used another "reporter" that was providing debugging capabilities.

I don't think it makes sense to support plural WithCollectorProxy(), because (a) they will never be used for sampling config, so the CollectorProxy interface is too wide, and (b) having just WithReporter() makes more sense anyway given that we don't know the purpose of the reporter and cannot make assumptions about it being implemented as a proxy.

Reply:
Can you please describe why do you use multiple reporters? 

Note that `CollectorProxy` is added in `build(primaryProxy, m, l)` and also as `withCollectorProxy`. Sampling and baggage manager is used only from the `primaryProxy`. I think this coupling makes sense bc manager is not optional and uses the same transport as reporter. Do you think we will use grpc reporter and tchannel for manager? It still could use this interface - the mains would look simpler.

Reply:
I have changed this method signature to `withReporter(reporter.Reporter)` but keeping the build with collector proxy. Does it work for you?

Reply:
yep, that's what I meant. We have two cases for additional reports: (1) cloning the traffic to another DC via another tchannel reporter, and (b) sending all received spans to a fake reporter (based on https://github.com/uber-go/gwr) that runs in process and allows attaching listeners at runtime, either for debugging purposes, or for getting a stream of spans for some other type of analysis (conceptually similar to z-pages in OpenCensus).